src/main/java/cms/gov/madie/measure/services/QdmPackageService.java

```diff
@@ -45,17 +47,29 @@ public PackageDto getMeasurePackage(Measure measure, String accessToken) {
     headers.set(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE);
     HttpEntity<Measure> entity = new HttpEntity<>(measure, headers);
     try {
-      log.info("requesting measure package for measure [{}] from qdm service", measure.getId());
+      log.info("Requesting measure package for measure [{}] from QDM service", measure.getId());
       byte[] exportPackage =
           qdmServiceRestTemplate.exchange(uri, HttpMethod.PUT, entity, byte[].class).getBody();
+
       return PackageDto.builder().exportPackage(exportPackage).fromStorage(false).build();
+
+    } catch (HttpClientErrorException ex) {
+      String errorMessage = ex.getResponseBodyAsString();
+
+      log.error("Error from QDM service for measure [{}]: {}", measure.getId(), errorMessage);
+      if (ex.getMessage().contains("HQMF")) {
+        throw new HQMFServiceException();
+      }
+      throw new InternalServerException("QDM service error: " + errorMessage);
```

Comment:
my guess is this may not be required if you remove the try catch in qdm service- https://github.com/MeasureAuthoringTool/madie-qdm-service/pull/98#discussion_r1806922693